Source/WebCore/rendering/RenderElement.cpp

```diff
@@ -865,6 +873,9 @@ void RenderElement::styleWillChange(StyleDifference diff, const RenderStyle& new
         setHasNonVisibleOverflow(false);
         setHasTransformRelatedProperty(false);
         setHasReflection(false);
+    } else {
+        if (newStyle.contentVisibility() == ContentVisibility::Auto && element())
+            ContentVisibilityDocumentState::observe(*element());
```

Comment:
It'd be great to find a common place for ContentVisibilityDocumentState::observer/unobserve logic here in styleWillChange

Reply:
Good point, I did not like it either. I added a lambda for this.